haystack/preview/ready_made_pipelines/rag.py

```diff
@@ -0,0 +1,97 @@
+from typing import Optional, List
+
+from haystack.preview import Pipeline
+from haystack.preview.dataclasses import Answer
+from haystack.preview.document_stores import InMemoryDocumentStore
+from haystack.preview.components.retrievers import InMemoryBM25Retriever, InMemoryEmbeddingRetriever
+from haystack.preview.components.embedders import SentenceTransformersTextEmbedder
+from haystack.preview.components.generators import GPTGenerator
+from haystack.preview.components.builders.answer_builder import AnswerBuilder
+from haystack.preview.components.builders.prompt_builder import PromptBuilder
+
+
+class RAGPipeline:
+    """
+    A simple ready-made pipeline for RAG. It requires a populated document store.
+
+    If an embedding model is given, it uses embedding retrieval. Otherwise it falls back to BM25 retrieval.
+
+    Example usage:
+
+    ```python
+    rag_pipe = RAGPipeline(document_store=InMemoryDocumentStore())
+    answers = rag_pipe.run(query="Who lives in Rome?")
+    >>> Answer(data="Giorgio")
+    ```
+
+    """
+
+    def __init__(
+        self,
+        document_store: InMemoryDocumentStore,
+        generation_model: str = "gpt-3.5-turbo",
+        prompt_template: Optional[str] = None,
+        embedding_model: Optional[str] = None,
+    ):
+        """
+        :param document_store: An instance of a DocumentStore to retrieve documents from.
+        :param generation_model: The name of the model to use for generation.
+        :param prompt_template: The template to use for the prompt. If not given, a default template is used.
+        :param embedding_model: The name of the model to use for embedding. If not given, BM25 is used.
+        """
+        prompt_template = (
```

Comment:
please use prompt_template from init if it is specified

Reply:
Line 43 does precisely that...